src/go/rpk/pkg/cli/cmd/status.go

```diff
@@ -149,44 +153,45 @@ func getMetrics(
 	timeout time.Duration,
 	conf config.Config,
 	send bool,
-	out chan<- [][]string,
-) {
-	rows := [][]string{}
+) (*metricsResult, error) {
+	res := &metricsResult{[][]string{}, nil}
 	osInfo, err := system.UnameAndDistro(timeout)
 	if err != nil {
-		log.Info("Error querying OS info: ", err)
-	} else {
-		rows = append(rows, []string{"OS", osInfo})
+		return res, fmt.Errorf("Error querying OS info: %v", err)
```

Comment:
I liked it before, where individual metrics failures didn't prevent others from being gathered. Further down there's an `errs`, perhaps all errors could be gathered there?

Reply:
The OS info could be retrieved separately. I will do that.